src/Common/HashTable/HashTable.h

```diff
@@ -617,6 +617,163 @@ class HashTable : private boost::noncopyable,
         }
     }
 
+    // Prefetching keys will reduce cache misses and improve performance.
+    template <typename Derived, bool is_const>
+    class prefetching_iterator_base
+    {
+
+        using Container = std::conditional_t<is_const, const Self, Self>;
+        using cell_type = std::conditional_t<is_const, const Cell, Cell>;
+
+        struct RingBuffer
```

Comment:
This looks too complex for the task. Why won't we just prefetch `current + prefetch_ahead` if it is not zero?

Reply:
I want to optimize cell checking to avoid duplicates